src/pasta/Pallas.sol

```diff
@@ -486,23 +477,45 @@ library Pallas {
         return point.y < P_MOD / 2;
     }
 
-    // @dev Perform a modular exponentiation.
-    // @return base^exponent (mod modulus)
-    // This method is ideal for small exponents (~64 bits or less), as it is cheaper than using the pow precompile
-    // @notice credit: credit: Aztec, Spilsbury Holdings Ltd
-    function powSmall(uint256 base, uint256 exponent, uint256 modulus) internal pure returns (uint256) {
-        uint256 result = 1;
-        uint256 input = base;
-        uint256 count = 1;
+    function fromBytes(bytes32 compressed_x_coord) public view returns (PallasAffinePoint memory point) {
+        uint8 y_sign = uint8(compressed_x_coord[31]) >> 7;
+
+        uint256 x_coord;
+
+        x_coord += uint256(uint8(compressed_x_coord[31]) & 0x7F);
```

Comment:
Nit: this is the second time you compute `uint8(compressed_x_coord[31]) & 0x7F` no?

Reply:
Earlier we calculate `uint8(compressed_x_coord[31]) >> 7`, but I'm not sure re-using the `uint8(compressed_x_coord[31])` calculation would save an appreciable amount of gas.